### PR TITLE
allow localisation to be saved to git

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

**Comment by miltontony** [comment]
@smn ready for review

File: cms/models.py [hunk 1]
```diff
@@ -519,3 +519,45 @@ def auto_delete_category_to_git(sender, instance, **kwargs):
...
+
+@receiver(post_save, sender=Localisation)
+def auto_save_localisation_to_git(sender, instance, created, **kwargs):
+
+    data = {
+        "locale": instance.get_code(),
+    }
+
+    # TODO: Not yet implemented
+    # author = utils.get_author_from_user(instance.last_author)
```
Comment (smn) [inline cms/models.py]: remove?

File: cms/models.py [hunk 2]
```diff
@@ -519,3 +519,45 @@ def auto_delete_category_to_git(sender, instance, **kwargs):
...
+
+    workspace = EG.workspace(settings.GIT_REPO_PATH,
+                             index_prefix=settings.ELASTIC_GIT_INDEX_PREFIX)
+    try:
+        # FIXME: This can fail if we ever store a value with None
+        #        as the uuid.
```
Comment (smn) [inline cms/models.py]: EG prevents this from happening.

File: cms/models.py (hunk 1 above)
Comment (miltontony) [inline cms/models.py]: @smn I was just following what's there. In case we ever get around to setting the author

File: cms/models.py [hunk 3]
```diff
@@ -519,3 +519,45 @@ def auto_delete_category_to_git(sender, instance, **kwargs):
...
+        [localisation] = workspace.S(
+            eg_models.Localisation).filter(locale=instance.get_code())
+        original = localisation.get_object()
+        updated = original.update(data)
+        workspace.save(updated, 'Localisation updated: %s' % str(instance))
```
Comment (smn) [inline cms/models.py]: `str()` can fail because we may be given Unicode content like `Zoë`

File: cms/models.py [hunk 4]
```diff
@@ -519,3 +519,45 @@ def auto_delete_category_to_git(sender, instance, **kwargs):
...
+        workspace.refresh_index()
+    except (GitCommandError, ValueError), e:
+        print 'we got an error ', e
```
Comment (smn) [inline cms/models.py]: stray print

File: cms/models.py [hunk 5]
```diff
@@ -519,3 +519,45 @@ def auto_delete_category_to_git(sender, instance, **kwargs):
...
+        localisation = eg_models.Localisation(data)
+        workspace.save(
+            localisation, 'Localisation created: %s' % str(instance))
```
Comment (smn) [inline cms/models.py]: same here, assuming this is a copy & paste bug.

File: cms/models.py [hunk 6]
```diff
@@ -519,3 +519,45 @@ def auto_delete_category_to_git(sender, instance, **kwargs):
...
+        workspace.refresh_index()
+
+
+@receiver(post_delete, sender=Localisation)
+def auto_delete_localisation_to_git(sender, instance, **kwargs):
+    # TODO: Not yet implemented
+    # author = utils.get_author_from_user(instance.last_author)
```
Comment (smn) [inline cms/models.py]: can go I think

File: cms/models.py [hunk 7]
```diff
@@ -519,3 +519,45 @@ def auto_delete_category_to_git(sender, instance, **kwargs):
...
+    workspace = EG.workspace(settings.GIT_REPO_PATH,
+                             index_prefix=settings.ELASTIC_GIT_INDEX_PREFIX)
+    [localisation] = workspace.S(
+        eg_models.Localisation).filter(locale=instance.get_code())
+    original = localisation.get_object()
+    workspace.delete(original, 'Localisation deleted: %s' % str(instance))
```
Comment (smn) [inline cms/models.py]: also `str()`

**Comment by smn** [comment]
:+1: in general but perhaps we'd need to rename `language` in `Page` & `Category` to `locale` as well for consistency.

File: cms/models.py (hunk 3 above)
Comment (miltontony) [inline cms/models.py]: `unicode()` instead?

**Comment by miltontony** [comment]
Ok, most fixed in #44 (I was in the wrong branch and didn't check)

**Pull request merged**